cv-data-service-library/src/main/java/com/trihydro/library/helpers/TimGenerationHelper.java

```diff
@@ -447,67 +471,73 @@ public List<ResubmitTimException> resetTimStartTimeAndResubmitToOde(List<Long> a
+                        tum.getActiveTimId());
                     utility.logWithDate(exMsg);
                     exceptions.add(new ResubmitTimException(activeTimId, exMsg));
                     continue;
                 }
-                Milepost firstPoint = allMps.get(0);
-                Milepost secondPoint = allMps.get(1);
+
+                Milepost anchorMp = attemptToCalculateAnchorPoint(tum, exceptions, allMps);
+                if (anchorMp == null) {
+                    return null;
+                }
 
                 // reduce the mileposts by removing straight away posts
-                var anchorMp = getAnchorPoint(firstPoint, secondPoint);
-                reduced_mps = milepostReduction.applyMilepostReductionAlgorithm(allMps, config.getPathDistanceLimit());
-                OdeTravelerInformationMessage tim = getTim(tum, reduced_mps, allMps, anchorMp, true, false);
+                reduced_mps = milepostReduction.applyMilepostReductionAlgorithm(allMps,
+                    config.getPathDistanceLimit());
+                OdeTravelerInformationMessage tim =
+                    getTim(tum, reduced_mps, allMps, anchorMp, true, false);
                 if (tim == null) {
                     String exMsg = String.format("Failed to instantiate TIM for active_tim_id %d",
-                            tum.getActiveTimId());
+                        tum.getActiveTimId());
                     utility.logWithDate(exMsg);
                     exceptions.add(new ResubmitTimException(activeTimId, exMsg));
                     continue;
                 }
                 WydotTravelerInputData timToSend = new WydotTravelerInputData();
                 timToSend.setTim(tim);
                 var extraEx = sendTim(timToSend, tum, activeTimId, reduced_mps);
-                if (extraEx.size() > 0) {
+                if (!extraEx.isEmpty()) {
                     exceptions.addAll(extraEx);
                 }
             } catch (Exception ex) {
                 exceptions.add(new ResubmitTimException(activeTimId, ex.getMessage()));
             }
         }
-        if (exceptions.size() > 0) {
-            utility.logWithDate("Errors occurred while resubmitting TIMs: " + gson.toJson(exceptions),
-                    TimGenerationHelper.class);
+        if (!exceptions.isEmpty()) {
+            utility.logWithDate(
+                "Errors occurred while resubmitting TIMs: " + gson.toJson(exceptions),
+                TimGenerationHelper.class);
         }
         return exceptions;
     }
 
     /**
      * Expires existing TIMs and resubmits TIMs to the ODE
-     * 
+     *
      * @param activeTimIds TIMs to resubmit
-     * @return Errors that occurred while processing the request
      */
-    public List<ResubmitTimException> expireTimAndResubmitToOde(List<Long> activeTimIds) {
+    public void expireTimAndResubmitToOde(List<Long> activeTimIds) {
```

Comment:
Clarification (non-blocking): Since the return value of this method was unused, I changed the return type to void.